sdk/storage/azure-storage-blobs/src/append_blob_client.cpp

```diff
@@ -110,6 +110,13 @@ namespace Azure { namespace Storage { namespace Blobs {
       protocolLayerOptions.EncryptionAlgorithm = m_customerProvidedKey.Value().Algorithm.ToString();
     }
     protocolLayerOptions.EncryptionScope = m_encryptionScope;
+    if (options.ImmutabilityPolicy.HasValue())
+    {
+      protocolLayerOptions.ImmutabilityPolicyExpiry = options.ImmutabilityPolicy.Value().ExpiresOn;
+      protocolLayerOptions.ImmutabilityPolicyMode = options.ImmutabilityPolicy.Value().PolicyMode;
+    }
```

Comment:
```suggestion
   {
      auto const& immutabilityPolicy = options.ImmutabilityPolicy;
      if (immutabilityPolicy)
      {
        protocolLayerOptions.ImmutabilityPolicyExpiry = immutabilityPolicy->ExpiresOn;
        protocolLayerOptions.ImmutabilityPolicyMode = immutabilityPolicy->PolicyMode;
      }
    }
```

Reply:
Wow, since when do we support `->` for nullables! I didn't know that. I think this piece of code can be further optimized to

```C++
if (options.ImmutabilityPolicy) {
  protocolLayerOptions.ImmutabilityPolicyExpiry = options.ImmutabilityPolicy->ExpiresOn;
  protocolLayerOptions.ImmutabilityPolicyMode = options.ImmutabilityPolicy->PolicyMode;
}
```

It's only 4 lines of code.